vortex-array/src/arrays/chunked/canonical.rs

```diff
@@ -17,7 +17,9 @@ impl ArrayCanonicalImpl for ChunkedArray {
             return Ok(Canonical::empty(self.dtype()));
         }
         if self.nchunks() == 1 {
-            return self.chunks()[0].to_canonical();
+            let chunk = &self.chunks()[0];
+            let cast_chunk = cast(chunk, &self.dtype)?;
+            return cast_chunk.to_canonical();
```

Comment:
huh? ChunkedArray validates that the dtype is the same when being built. I think we have a bug where we call new_unchecked with wrong dtypes

Reply:
must be something like that, because I manage to fail here in a spiral test (even with this I get weird behavior, so I'm now trying to root cause the original issue)

Reply:
the difference I see here is in nullability - the chunk is nonnullable while the array is.

Reply:
I agree with rob. I don't think we should cast here.

Reply:
Yeah I found the spiral bug, tomorrow I'll push here some things that can/should help in the future.